packages/utils/src/DataUtils.ts

```diff
@@ -0,0 +1,5 @@
+export const EMPTY_ARRAY = Object.freeze([]);
+
+export const EMPTY_MAP = new Map();
```

Comment:
I'll type it as `ReadonlyMap<never, never>`. I'm not sure how I feel about modifying the `.set` function...

Reply:
My concern is if it's consumed in JS and then modified, anywhere else that imported `EMPTY_MAP` now no longer has an empty map. Since you can freeze an array and prevent modification that works, but you can't freeze a Map afaik. It still lets you set values.

Since this is intended to be a re-used constant in multiple places for a Map, I think we should ensure it's always empty at runtime